### PR TITLE
i18n(it): translate `core-concepts`

### PR DESCRIPTION
**Pull request by itsmatteomanf**
#### What kind of changes does this PR include?

- Translated content

#### Description

Creates translations for the `core-concepts` section of the docs.

- [x] [astro-components.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/it/core-concepts/astro-components.mdx)
- [x] [astro-pages.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/it/core-concepts/astro-pages.mdx)
- [x] [astro-syntax.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/it/core-concepts/astro-syntax.mdx)
- [x] [endpoints.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/it/core-concepts/endpoints.mdx)

### CODE REVIEW

File: src/content/docs/it/core-concepts/astro-components.mdx [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+- Tutti i tag `<script>` vengono lasciati non raggruppati, trattati come se avessero `is:inline`.
+- Possono solo [fare riferimento alle risorse che si trovano nella cartella `public/`](/it/core-concepts/project-structure/#public).
+
+:::note
+Un [elemento `<slot />`](/it/core-concepts/astro-components/#slot) all'interno di un componente HTML funzionerà come in un componente Astro. Per utilizzare invece l'elemento [HTML Web Component Slot](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot), aggiungi `is:inline` al tuo `<slot >` elemento.
+:::
+
+## Prossimi passi
+
+📚 Scopri come utilizzare i [componenti del framework UI](/it/core-concepts/framework-components/) nel tuo progetto Astro.
```
Comment (csarnataro) [inline src/content/docs/it/core-concepts/astro-components.mdx:347]: I think here - and elsewhere - `framework components` means `other framework components`, like React, Svelte, etc.
Hence I would not use `componenti del framework` which makes you think to Astro itself.
I would use `componenti di framework UI`.

File: src/content/docs/it/core-concepts/astro-components.mdx [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+<!-- Commenti HTML supportati! -->
+{/*Anche la sintassi dei commenti JS è valida! */}
+
+<Banner />
+<h1>Hello, world!</h1>
+
+<!-- Utilizza proprietà e altre variabili dallo script del componente: -->
+<p>{title}</p>
+
+<!-- Includi altri componenti del framework dell'interfaccia utente con una direttiva `client:` per idratare: -->
```
Comment (csarnataro) [inline src/content/docs/it/core-concepts/astro-components.mdx:83]:
```suggestion
<!-- Includi componenti di altri framework UI con una direttiva `client:` per idratare: -->
```

See my comment at line 347 in 
`src/content/docs/it/core-concepts/astro-components.mdx`

File: src/content/docs/it/core-concepts/astro-components.mdx [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+</Wrapper>
+```
+
+
+Utilizza un attributo `slot="my-slot"` sull'elemento figlio che desideri passare a un segnaposto `<slot name="my-slot" />` corrispondente nel tuo componente.
+
+Tieni presente che gli slot denominati devono essere figli immediati del componente. Non è possibile passare slot con nome attraverso elementi nidificati.
+
+:::tip
+Gli slot con nome possono anche essere passati a [componenti del framework UI](/it/core-concepts/framework-components/)!
```
Comment (csarnataro) [inline src/content/docs/it/core-concepts/astro-components.mdx:253]:
```suggestion
Gli slot con nome possono anche essere passati a [componenti di altri framework UI](/it/core-concepts/framework-components/)!
```
See my comment below.

File: src/content/docs/it/core-concepts/astro-pages.mdx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+---
+// Example: src/pages/index.astro
+import MySiteLayout from '../layouts/MySiteLayout.astro';
+---
+<MySiteLayout>
+  <p>My page content, wrapped in a layout!</p>
+</MySiteLayout>
+```
+
+📚 Leggi di più sui [componenti del layout](/it/core-concepts/layouts/) in Astro.
```
Comment (csarnataro) [inline src/content/docs/it/core-concepts/astro-pages.mdx:59]:
```suggestion
📚 Leggi di più sui [componenti di layout](/it/core-concepts/layouts/) in Astro.
```

File: src/content/docs/it/core-concepts/astro-pages.mdx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+# Title
+
+This is my page, written in **Markdown.**
+```
+
+📚 Ulteriori informazioni su [Markdown](/it/guides/markdown-content/) in Astro.
+
+## Pagine HTML
+
+I file con estensione `.html` possono essere inseriti in `src/pages/` e utilizzati direttamente come pagine del tuo sito. Tieni presente che alcune funzionalità chiave di Astro non sono supportate in [Componenti HTML](/it/core-concepts/astro-components/#componenti-html).
```
Comment (csarnataro) [inline src/content/docs/it/core-concepts/astro-pages.mdx:84]:
```suggestion
I file con estensione `.html` possono essere inseriti in `src/pages/` e utilizzati direttamente come pagine del tuo sito. Tieni presente che alcune funzionalità chiave di Astro non sono supportate nei [componenti HTML](/it/core-concepts/astro-components/#componenti-html).
```

File: src/content/docs/it/core-concepts/astro-pages.mdx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+## Pagina di errore 404 personalizzata
+
+Per una pagina di errore 404 personalizzata, puoi creare un file `404.astro` o `404.md` in `/src/pages`.
+
+Questo creerà una pagina `404.html`. La maggior parte dei [servizi di distribuzione](/it/guides/deploy/) la troveranno e la utilizzeranno.
```
Comment (csarnataro) [inline src/content/docs/it/core-concepts/astro-pages.mdx:90]: I'm not sure about `servizi di distribuzione`, usually in Italian we use this term for gas companies.
I would use something else, like `fornitori di hosting` or `fornitori di servizi cloud`. Any other idea?

**Review by csarnataro — COMMENTED** [review]
I left a few comment on the first 2 pages. I will continue with the rest of the pages in the next days.

File: src/content/docs/it/core-concepts/astro-pages.mdx (hunk 5 above)
Comment (itsmatteomanf) [reply src/content/docs/it/core-concepts/astro-pages.mdx:84]: The capitalisation existed in the original, which is why I left it...

File: src/content/docs/it/core-concepts/astro-pages.mdx (hunk 6 above)
Comment (itsmatteomanf) [reply src/content/docs/it/core-concepts/astro-pages.mdx:90]: Yeah, we could merge your two suggestions: `fornitori di servizi di hosting`.

```suggestion
Questo creerà una pagina `404.html`. La maggior parte dei [fornitori di servizi di hosting](/it/guides/deploy/) la troveranno e la utilizzeranno.
```

File: src/content/docs/it/core-concepts/astro-components.mdx [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+
+I componenti Astro sono estremamente flessibili. Spesso, un componente Astro conterrà alcune **UI riutilizzabili sulla pagina**, come un'intestazione o una scheda del profilo. Altre volte, un componente Astro può contenere un frammento di codice HTML più piccolo, come una raccolta di tag `<meta>` comuni che semplificano il lavoro con la SEO. I componenti Astro possono contenere anche un intero layout di pagina.
+
+La cosa più importante da sapere sui componenti Astro è che **non vengono renderizzati sul client**. Eseguono il rendering in HTML in fase di compilazione o su richiesta utilizzando il [rendering lato server (SSR)](/it/guides/server-side-rendering/). Puoi includere il codice JavaScript all'interno del frontmatter del tuo componente e tutto verrà rimosso dalla pagina finale inviata ai browser dei tuoi utenti. Il risultato è un sito più veloce, con zero impatto JavaScript aggiunto per impostazione predefinita.
+
+Quando il tuo componente Astro necessita di interattività lato client, puoi aggiungere [tag HTML standard `<script>`](/it/guides/client-side-scripts/) o [componenti UI Framework](/it/core-concepts /componenti-strutturali/#componenti-interattivi-idratanti).
```
[the remark this answers is not in the document]
Comment (itsmatteomanf) [reply src/content/docs/it/core-concepts/astro-components.mdx:12]: Should probably remain untranslated, fully, though...

```suggestion
Quando il tuo componente Astro necessita di interattività lato client, puoi aggiungere [tag HTML standard `<script>`](/it/guides/client-side-scripts/) o [componenti UI Framework](/it/core-concepts/framework-components/#hydrating-interactive-components).
```

File: src/content/docs/it/core-concepts/astro-components.mdx [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+<Banner />
+<h1>Hello, world!</h1>
+
+<!-- Utilizza proprietà e altre variabili dallo script del componente: -->
+<p>{title}</p>
+
+<!-- Includi altri componenti del framework dell'interfaccia utente con una direttiva `client:` per idratare: -->
+<ReactPokemonComponent client:visible />
+
+<!-- Mescola HTML con espressioni JavaScript, simili a JSX: -->
```
[the remark this answers is not in the document]
Comment (itsmatteomanf) [reply src/content/docs/it/core-concepts/astro-components.mdx:86]: Like it.

```suggestion
<!-- Usa espressioni JavaScript all'interno di tag HTML, come in JSX: -->
```

File: src/content/docs/it/core-concepts/astro-components.mdx (hunk 1 above)
Comment (itsmatteomanf) [reply src/content/docs/it/core-concepts/astro-components.mdx:347]: Gonna do a check if there are other places...

```suggestion
📚 Scopri come utilizzare i [componenti di framework UI](/it/core-concepts/framework-components/) nel tuo progetto Astro.
```

**Review by itsmatteomanf — COMMENTED** [review]
Looked over your changes @csarnataro, thank you!

**Comment by ElianCodes** [comment]
Hey there @itsmatteomanf! There are some unresolved suggestions here. Can you take a look at them whenever you have time, so we can get this merged?

Thanks ❤️

**Comment by itsmatteomanf** [comment]
Should be all done, @ElianCodes :)

**Comment by ElianCodes** [comment]
Taking a look now 👀

**Review by ElianCodes — APPROVED** [review]
Looks good to me! Thanks so much for putting the work into this @itsmatteomanf!

**Pull request merged**